include/maths/CBoostedTreeLoss.h

```diff
@@ -488,6 +568,66 @@ class MATHS_EXPORT CMsle final : public CLoss {
     CArgMinLoss minimizer(double lambda, const CPRNG::CXorOShiro128Plus& rng) const override;
     const std::string& name() const override;
     bool isRegression() const override;
+
+    void acceptPersistInserter(core::CStatePersistInserter& inserter) const override;
+    bool acceptRestoreTraverser(core::CStateRestoreTraverser& traverser) override;
+
+private:
+    double m_Offset;
+};
+
+//! \brief The pseudo-Huber loss function.
+//!
+//! DESCRIPTION:\n
+//! Formally, the pseudo-Huber loss definition we use is
+//! \f$\delta^2 (\sqrt{1 + \frac{(a - p)^2}{\delta^2}} - 1)\f$.
+//! However, we approximate this by a quadratic form which has its minimum p = a and
+//! matches the value and derivative of the pseudo-Huber loss function. For example, if the
+//! current prediction for the i'th training point is \f$p_i\f$, the loss is defined
+//! as
+//!   \f[
+//!     l_i(p) =  \delta^{2} \left(\sqrt{1 + \frac{\left(a_i - p_i\right)^{2}}{\delta^{2}}} - 1\right)
+//! + \frac{- a_i + p_i}{\sqrt{\frac{\delta^{2} + \left(a_i - p_i\right)^{2}}{\delta^{2}}}}(p-p_i)
+//! + \frac{- a_i + p_i}{2\sqrt{\frac{\delta^{2} + \left(a_i - p_i\right)^{2}}{\delta^{2}}}\left(a_i-p_i\right)}(p-p_i)^2
+//! \f]
+//! For this approximation we compute first and second derivative (gradient and curvature)
+//! with respect to p and then substitute p=p_i.
+//! As  result we obtain the following formulas for the gradient:
```

Comment:
```suggestion
//! As a result we obtain the following formulas for the gradient:
```

Reply:
Fixed now, sorry missed it on the first walk through.